agent/host_agent_suite_test.go

```diff
@@ -110,3 +132,42 @@ func writeKubeConfig() {
 	_, err = getKubeConfig().Write(kubeConfigData)
 	Expect(err).NotTo(HaveOccurred())
 }
+
+func setupTestInfra(ctx context.Context, hostname, kubeconfig string, namespace *corev1.Namespace) *e2e.ByoHostRunner {
```

Comment:
Wondering if this is required. Can we initialize `byoHostRunner` in the `BeforeSuite` itself? Then things like `PathToHostAgentBinary`, `ByoHostName` and `Namespace` can be from inside each `BeforeEach`. wdyt?

Reply:
I wondered about the same, but some of the `CommandArgs` passed during the tests are contradictory, and not properly clearing them up can lead to unpredictable behavior. For example, if the `--skip-installation` flag test executes before the k8s installation test and the `--skip-installation` flag is not cleared from the object, it can cause the k8s install test to fail.